### PR TITLE
fix: add missing validations in wizard

### LINKED ISSUE #3826: bug: Some validations are not working in wizard Steps.
**Describe the bug**
Country Field validation is not working in Wizard `basic details step`. 
Microlocations field validation not working in `session Speakers form`

**To Reproduce**
Steps to reproduce the behaviour:
1. Create an event.
2. Add a paid ticket.
3.Levae the country option blank.
4. Publish the event.
5. See Error.

**Expected behaviour**
The validation should work properly.

**Additional context**

I will work on this issue.

### PR DESCRIPTION
**Pull request by snitin315**
Fixes #3826 

#### Short description of what this resolves:
Added validation for country and microlocation fields in wizard.

#### Changes proposed in this pull request:
Screenshot from 2020-01-21 23-08-25

Screenshot from 2020-01-21 23-08-38

Screenshot from 2020-01-21 23-09-34

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/modals/tax-info-modal.hbs [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <div class="field">
       <label class="required">{{t 'Choose country'}}</label>
       {{#ui-dropdown class='search selection' selected=tax.country forceSelection=false fullTextSearch=true}}
-        {{input type='hidden' id='tax_country' value=tax.country}}
+        {{input type='hidden' id='tax_country' name='tax_country' value=tax.country}}
```
Comment (kushthedude) [inline app/templates/components/modals/tax-info-modal.hbs]: name not required explicitly, id will do

File: app/components/forms/wizard/sessions-speakers-step.js [hunk 2]
```diff
@@ -86,6 +86,24 @@ export default Component.extend(EventWizardMixin, FormMixin, {
...
+          identifier : 'microlocation',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.l10n.t('Please enter name for microlocation')
+            }
+          ]
+        },
+        microlocationFloor: {
+          identifier : 'microlocation_floor',
```
Comment (kushthedude) [inline app/components/forms/wizard/sessions-speakers-step.js]: Make it a number validation

File: app/components/forms/wizard/sessions-speakers-step.js [hunk 3]
```diff
@@ -86,6 +86,28 @@ export default Component.extend(EventWizardMixin, FormMixin, {
...
+          rules      : [
+            {
+              type   : 'empty',
```
Comment (kushthedude) [inline app/components/forms/wizard/sessions-speakers-step.js]: No need for this then !

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 4]
```diff
@@ -260,7 +260,7 @@
           <label class="required">{{t 'Select country'}}</label>
           {{#ui-dropdown class='search selection' selected=data.event.paymentCountry forceSelection=false
                          fullTextSearch=true}}
-            {{input type='hidden' id='payment_country' value=data.event.paymentCountry}}
+            {{input type='hidden' id="payment_country" value=data.event.paymentCountry}}
```
Comment (maze-runnar) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: why ??

File: app/components/forms/wizard/sessions-speakers-step.js (hunk 3 above)
Comment (snitin315) [reply app/components/forms/wizard/sessions-speakers-step.js]: Its not working without empty. 

Screenshot from 2020-01-22 17-23-52

Comment (snitin315) [reply app/components/forms/wizard/sessions-speakers-step.js]: @kushthedude  `tax_rate` is also implemented as this. 

Screenshot from 2020-01-22 17-26-14

File: app/templates/components/forms/wizard/basic-details-step.hbs (hunk 4 above)
Comment (snitin315) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: It's a typo. I will fix it.

File: app/templates/components/forms/wizard/sessions-speakers-step.hbs [hunk 5]
```diff
@@ -61,11 +61,11 @@
       {{#each microlocations as |microlocation|}}
         <div class="{{if device.isMobile 'grouped'}} fields">
           <div class="{{unless device.isMobile 'three wide'}} field">
-            {{input type='text' value=microlocation.name placeholder='Name'}}
+            {{input type='text' id="microlocation" value=microlocation.name placeholder='Name'}}
           </div>
           <div class="{{unless device.isMobile 'four wide'}} field">
             <div class="ui action input fluid">
-              {{input type='number' name='microlocation' value=microlocation.floor placeholder='Floor'}}
+              {{input type='number' id='microlocation_floor' value=microlocation.floor placeholder='Floor'}}
```
Comment (kushthedude) [inline app/templates/components/forms/wizard/sessions-speakers-step.hbs]: Alright then no need for number validation, I didn't saw the input type is already defined as `number`.

Comment (snitin315) [reply app/templates/components/forms/wizard/sessions-speakers-step.hbs]: @kushthedude Done .

**Pull request merged**